stko/calculators/macromodel_calculators.py

```diff
@@ -0,0 +1,250 @@
+            The full path to the MacroModel executable.
+
+        output_dir : :class:`str`
+            The path to the output directory.
+
+        timeout : :class:`int`
+            The number of seconds to wait for the calculation to
+            complete.
+
+        forcefield : :class:`int`
+            The force field to use.
+            Force field arguments can be the following:
+            +------------+------------+
+            |  1  | MM2               |
+            +------------+------------+
+            |  2  | MM3               |
+            +------------+------------+
+            |  3  | AMBER             |
+            +------------+------------+
+            |  4  | AMBER94           |
+            +------------+------------+
+            |  5  | OPLSA             |
+            +------------+------------+
+            |  10 | MMFF94 and MMFF94s|
+            +------------+------------+
+            |  14 | OPLS_2005         |
+            +------------+------------+
+            |  16 | OPLS3e            |
+            +------------+------------+
+
+        """
+        super().__init__(
+            macromodel_path=macromodel_path, output_dir=output_dir,
+            timeout=timeout,
+            force_field=force_field,
+        )
+        self._macromodel_path = macromodel_path
+        self._output_dir = output_dir
+        self._timeout = timeout
+        self._forcefield = force_field
+
+    def get_results(self, mol):
+        """
+        Calculate the energy of `mol`.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to calculate the energy of.
+
+        Returns
+        -------
+        :class:`.EnergyResults`
+            The energy and units of the energy.
+
+        """
+
+        return EnergyResults(
+            generator=self.calculate(mol),
+            unit_string='kJ mol-1'
+        )
+
+    def calculate(self, mol):
+        """
+        Performs the calculation.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to calculate the energy of.
+
+        """
+        run_name = str(uuid4().int)
+        if self._output_dir is None:
+            output_dir = run_name
+        else:
+            output_dir = self._output_dir
+        mol_path = f'{run_name}.mol'
+        mae_path = f'{run_name}.mae'
+        # First write the molecule
+        mol.write(mol_path)
+        # MacroModel requires a `.mae` file
+        self._run_structconvert(mol_path, mae_path)
+        # Generate the `.com` file
+        self._generate_com(mol, run_name)
+        # Run the calculation
+        self._run_bmin(mol, run_name)
+        # Convert `.maegz` output to `.mae`
+        self._convert_maegz_to_mae(run_name)
+        # Read the results
+        result = self._read_mmo(run_name)
+        move_generated_macromodel_files(run_name, output_dir)
+        yield result
+
+    @staticmethod
+    def _read_mmo(run_name):
```

Comment:
This is the remit of an extractor. Not fussed if it stays here to be fair because this requires much less flexibility than xtb or orca - but if you wanted to get other properties, you might want to put this in an extractor.

Reply:
Your choice.